### PR TITLE
chore: Fix more Bash inconsistencies

### PR DESCRIPTION
**Pull request by hyperupcall**
This mainly fixes the following:
- Not only are more path-related variables quoted (ie. `git-touch`, `git-scp`), but variables like `$BRANCH` are.
- Add more `-r` to `read` invocations
- Stuff like `$(echo $(printf '%s\n' 'thing'))` is changed to `$(printf '%s\n' 'thing')`

### CODE REVIEW

File: bin/git-psykorebase [hunk 1]
```diff
@@ -19,7 +19,7 @@ function usage()
     echo "  -c|--continue:		Continue after the user updates conflicts."
 }
 
-while [[ $# > 0 ]]
+while [[ $# -gt 0 ]]
```
Comment (guenhter) [inline bin/git-psykorebase:22]: Why was this changed? `>` is be perfectly and expressive find within `[[ ... ]]`, isn't it?

Comment (hyperupcall) [reply bin/git-psykorebase:22]: `>` and `<` perform [lexicographical comparison](https://www.shellcheck.net/wiki/SC2071) rather than a numerical comparison in this context, so it was changed to `-gt`

Comment (guenhter) [reply bin/git-psykorebase:22]: I see. Thx for the explanation. With this new knowledge I now have to have a look at some of my own scripts as well :)

**Review by spacewander — APPROVED** [review]
